File: trustdevice/src/main/java/cn/tongdun/mobrisk/providers/sensor/SensorInfoRawData.java
package cn.tongdun.mobrisk.providers.sensor;

import android.util.Pair;

import java.util.ArrayList;
import java.util.List;

import cn.tongdun.mobrisk.providers.RawData;

/**
 * @description: SensorInfoRawData
 * @author: wuzuchang
 * @date: 2022/12/6
 */
@Deprecated(since = "pro no such class")
public class SensorInfoRawData extends RawData<String> {
    private String sensorInfo;

    public SensorInfoRawData(String sensorInfo) {
        this.sensorInfo = sensorInfo;
    }

    private String[] getSensorInfo() {
        return sensorInfo.split(",");
    }

    @Override
    public List<Pair<String, String>> loadData() {
        List<Pair<String, String>> data = new ArrayList<>();
        for (String sensorInfo : getSensorInfo()) {
            String[] sensor = sensorInfo.split(":");
            if (sensor.length >= 2) {
                data.add(new Pair<>(sensor[0], sensor[1]));
            }
        }
        return data;
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/providers/SensorInfoRawData.kt
package cn.tongdun.mobrisk.providers

import android.text.TextUtils
import cn.tongdun.mobrisk.core.tools.executeSafe
import org.json.JSONArray

/**
 * @description:SensorInfoRawData
 * @author: wuzuchang
 * @date: 2023/5/16
 */
data class SensorInfoRawData(val sensorInfo: String) {
    fun getSensorTotal(): Int {
        if (TextUtils.isEmpty(sensorInfo)) {
            return 0
        }
        var sensorArray: JSONArray? = null
        executeSafe { sensorArray = JSONArray(sensorInfo) }
        return sensorArray?.length() ?: 0
    }
}